app/models/user.rb

```diff
@@ -738,8 +738,21 @@ def self.system_avatar_template(username)
 
   def self.letter_avatar_color(username)
     username ||= ""
-    color = LetterAvatar::COLORS[Digest::MD5.hexdigest(username)[0...15].to_i(16) % LetterAvatar::COLORS.length]
-    color.map { |c| c.to_s(16).rjust(2, '0') }.join
+    if SiteSetting.restrict_letter_avatar_colors.present?
+      hex_length = 6
+      colors = SiteSetting.restrict_letter_avatar_colors
+      length = colors.length / hex_length
```

Comment:
This won't work when there are more than 6 colors.

```suggestion
      length = colors.count("|") + 1
```